src/storage/mod.rs

```diff
@@ -488,15 +488,35 @@ impl<E: Engine> TestStorageBuilder<E> {
     }
 }
 
+/// `Storage` implements transactional KV APIs and raw KV APIs on a given `Engine`. An `Engine`
+/// provides low level KV functionality. `Engine` has multiple implementations. When a TiKV server
+/// is running, a `RaftKv` will be the underlying `Engine` of `Storage`. The other two types of
+/// engines are for test purpose.
+///
+/// `Storage` is actually a reference to storage. It is cloneable, and a clone of a `Storage` is
+/// actually a reference to the same storage, and is equivalent to the original one. It will be
+/// automatically stopped and destroyed when it and all its clones are dropped.
+///
+/// All read and write methods are performed on **a single region**, which is specified in the
+/// parameter `ctx`. However, `async_unsafe_destroy_range` is the only exception. It's performed on
+/// the whole TiKV.
+///
+/// Also note that the concept of *region* only exists when `Storage` is running with `RaftKv`. When
+/// running other kinds of `Engine`, all requests are executed on the whole storage.
 pub struct Storage<E: Engine> {
     // TODO: Too many Arcs, would be slow when clone.
     engine: E,
 
-    /// To schedule the execution of storage commands
+    /// To schedule the execution of storage commands.
```

Comment:
Actually I believe that it is used for only destroying the worker (so that it is kept as a field here).